kernels/alloc/Snakefile

```diff
@@ -0,0 +1,47 @@
+from util.snake.configs import get_snax_mac_config
+
+config = get_snax_mac_config()
+
+config["snaxoptflags"] = ",".join(
+    [
+        "dispatch-kernels",
+        "set-memory-space",
+        "set-memory-layout",
+        "realize-memref-casts",
+        "reuse-memref-allocs",
+        "insert-sync-barrier",
+        "dispatch-regions",
+        "linalg-to-library-call",
+        "snax-copy-to-dma",
+        "memref-to-snax",
+        "snax-to-func",
+        "clear-memory-space",
+    ]
+)
+
+
+module default_rules:
+    snakefile:
+        "../../util/snake/default_rules.smk"
+    config:
+        config
+
+
+rule all:
+    input:
+        "func.x",
+    shell:
+        "{config[vltsim]} {input[0]}"
+
+
+use rule * from default_rules exclude compile_simple_main as default_*
```

Comment:
what is this?

Reply:
This is to include/import rules from a different snakemake file declared as a module.
You use the module declaration to choose the file and to apply the configuration, and then those rules act as if they have been included in the file, but with the `default_` prefix.
In this case, a lot of rules are shared for many jobs, so instead of defining them everywhere multiple times, I import them from `default_rules.smk`

Reply:
but why exclude compile_simple_main?